src/mcts/search.cc

```diff
@@ -692,6 +689,8 @@ void SearchWorker::GatherMinibatch() {
       --minibatch_size;
       ++number_out_of_order;
     }
+    // Check for stop at the end so we have at least one node.
+    if (search_->stop_.load(std::memory_order_acquire)) return;
```

Comment:
I think this check should also be added above the continue for processing a collision.